src/ui/map.js

```diff
@@ -1185,49 +1190,53 @@ class Map extends Camera {
         } else if (typeof projection === 'string') {
             projection = (({name: projection}: any): ProjectionSpecification);
         }
-        return this._updateProjection(projection);
+        this._explicitProjection = projection;
+        return this._updateProjection();
     }
 
-    _updateProjection(explicitProjection?: ProjectionSpecification | null): this {
-        const prevProjection = this.getProjection();
-        if (explicitProjection === null) {
-            this._explicitProjection = null;
+    _updateProjectionTransition() {
+        // The projection isn't globe, we can skip updating the transition
+        if (this.getProjection().name !== 'globe') {
+            return;
         }
-        const projection = explicitProjection || this.getProjection();
 
+        const tr = this.transform;
+        const projection = tr.projection.name;
         let projectionHasChanged;
-        // At high zoom on globe, set transform projection to Mercator while _explicitProjection stays globe.
-        if (projection && projection.name === 'globe' && this.transform.zoom >= GLOBE_ZOOM_THRESHOLD_MAX) {
-            projectionHasChanged = this.transform.setProjection({name: 'mercator'});
-            this.transform.mercatorFromTransition = true;
-        } else {
-            projectionHasChanged = this.transform.setProjection(projection);
-            this.transform.mercatorFromTransition = false;
+
+        if (projection === 'globe' && tr.zoom >= GLOBE_ZOOM_THRESHOLD_MAX) {
+            projectionHasChanged = tr.setMercatorFromTransition();
+        } else if (projection === 'mercator' && tr.zoom < GLOBE_ZOOM_THRESHOLD_MAX) {
+            projectionHasChanged = tr.setProjection({name: 'globe'});
         }
 
-        // When called through setProjection, update _explicitProjection
-        if (explicitProjection) {
-            this._explicitProjection = (explicitProjection.name === "globe" ?
-                {name:'globe', center:[0, 0]} :
-                this.transform.getProjection());
+        if (projectionHasChanged) {
+            this.style.applyProjectionUpdate();
+            this.style._forceSymbolLayerUpdate();
+        }
+    }
+
+    _updateProjection(): this {
+        //check explicit projection separately as transform.projection hasn't been updated yet
+        const projection = this._explicitProjection || this.getProjection();
+        let projectionHasChanged;
+
+        if (projection.name === 'globe' && this.transform.zoom >= GLOBE_ZOOM_THRESHOLD_MAX) {
+            projectionHasChanged = this.transform.setMercatorFromTransition();
```

Comment:
Currently `setMercatorFromTransition` always returns true, but `transform.projection` hasn't changed in the case where we switch Mercator to Globe at high zoom. (This may be why we're seeing an extra call to`clearBackgroundTiles` in the unit test).

Should `transform.setMercatorFromTransition()` return whether the projection has updated similar to `transform.setProjection`?
https://github.com/mapbox/mapbox-gl-js/blob/4130912ec4fa859bae4d033479360c5679aa3db1/src/geo/transform.js#L250-L255

Reply:
Hey @SnailBones, can you elaborate on this? It's getting specifically called on a transition from globe -> mercator and not mercator -> globe. It's only called [here at map.js line 1204](https://github.com/mapbox/mapbox-gl-js/blob/a62035e9e18ae454e04d2cd637f4d3e11df01979/src/ui/map.js#L1204) and [here at map.js line 1221](https://github.com/mapbox/mapbox-gl-js/blob/a62035e9e18ae454e04d2cd637f4d3e11df01979/src/ui/map.js#L1221). I also checked that this proposed change would break toggling between mercator and globe at a high zoom. If I toggle globe -> mercator -> globe at a high zoom with setProjection and then zoom back out, this change doesn't register the change to back to globe projection and is still mercator.

Reply:
> Hey @SnailBones, can you elaborate on this?

When we change the style projection from Mercator to Globe at high zoom, we shouldn't clear tiles. (This is currently exacerbated by [unecessary calls to _updateProjection](https://github.com/mapbox/mapbox-gl-js/pull/12049#discussion_r916042902), in combination these result in clearing tiles on every style change while the projection is in globe by high zoom).

> I also checked that this proposed change would break toggling between mercator and globe at a high zoom. If I toggle globe -> mercator -> globe at a high zoom with setProjection and then zoom back out, this change doesn't register the change to back to globe projection and is still mercator.

Can you elaborate on what exactly you changed to see this behavior? I think that the correct behavior should result when this call in `_updateProjection` to be changed to:
```js
           projectionHasChanged = this.transform.setMercatorFromTransition();
```
while the call in `_updateProjectionTransition() should remain unaffected by the result:
```js
            tr.setMercatorFromTransition();
            projectionHasChanged = true;
```
but I may be wrong about that.

Reply:
OK I found a change that does cause the bug I noted above and does compare old projection and new projection. I'll add in a test case to make sure the we don't stumble into the bug I discovered above